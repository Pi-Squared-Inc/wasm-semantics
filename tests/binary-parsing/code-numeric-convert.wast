(module $mymodule
  (func (param i32) (result i32)

    i64.const 0
    i32.wrap_i64
    drop

    i32.const 0
    f32.convert_i32_s
    i32.trunc_f32_s
    drop

    i32.const 0
    f32.convert_i32_s
    i32.trunc_f32_u
    drop

    i32.const 0
    f64.convert_i32_s
    drop

    i32.const 0
    f64.convert_i32_s
    i32.trunc_f64_s
    drop

    i32.const 0
    f64.convert_i32_s
    i32.trunc_f64_u
    drop

    i32.const 0
    i64.extend_i32_s
    drop

    i32.const 0
    i64.extend_i32_u
    drop

    i32.const 0
    f32.convert_i32_s
    i64.trunc_f32_s
    drop

    i32.const 0
    f32.convert_i32_s
    i64.trunc_f32_u
    drop

    i32.const 0
    f64.convert_i32_s
    i64.trunc_f64_s
    drop

    i32.const 0
    f64.convert_i32_s
    i64.trunc_f64_u
    drop

    i32.const 0
    f32.convert_i32_s
    drop

    i32.const 0
    f32.convert_i32_u
    drop

    i64.const 0
    f32.convert_i64_s
    drop

    i64.const 0
    f32.convert_i64_u
    drop

    i32.const 0
    f64.convert_i32_s
    f32.demote_f64
    drop

    i32.const 0
    f64.convert_i32_s
    drop

    i32.const 0
    f64.convert_i32_u
    drop

    i64.const 0
    f64.convert_i64_s
    drop

    i64.const 0
    f64.convert_i64_u
    drop

    i32.const 0
    f32.convert_i32_s
    f64.promote_f32
    drop

    i32.const 0
    i32.extend8_s
    drop

    i32.const 0
    i32.extend16_s
    drop

    i64.const 0
    i64.extend8_s
    drop

    i64.const 0
    i64.extend16_s
    drop

    i64.const 0
    i64.extend32_s
    drop

    (return (i32.const 1))
  )
)
