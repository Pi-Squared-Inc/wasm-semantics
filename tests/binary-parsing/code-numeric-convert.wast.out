<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i64 . const 0  i32 . wrap_i64  drop  i32 . const 0  f32 . convert_i32_s  i32 . trunc_f32_s  drop  i32 . const 0  f32 . convert_i32_s  i32 . trunc_f32_u  drop  i32 . const 0  f64 . convert_i32_s  drop  i32 . const 0  f64 . convert_i32_s  i32 . trunc_f64_s  drop  i32 . const 0  f64 . convert_i32_s  i32 . trunc_f64_u  drop  i32 . const 0  i64 . extend_i32_s  drop  i32 . const 0  i64 . extend_i32_u  drop  i32 . const 0  f32 . convert_i32_s  i64 . trunc_f32_s  drop  i32 . const 0  f32 . convert_i32_s  i64 . trunc_f32_u  drop  i32 . const 0  f64 . convert_i32_s  i64 . trunc_f64_s  drop  i32 . const 0  f64 . convert_i32_s  i64 . trunc_f64_u  drop  i32 . const 0  f32 . convert_i32_s  drop  i32 . const 0  f32 . convert_i32_u  drop  i64 . const 0  f32 . convert_i64_s  drop  i64 . const 0  f32 . convert_i64_u  drop  i32 . const 0  f64 . convert_i32_s  f32 . demote_f64  drop  i32 . const 0  f64 . convert_i32_s  drop  i32 . const 0  f64 . convert_i32_u  drop  i64 . const 0  f64 . convert_i64_s  drop  i64 . const 0  f64 . convert_i64_u  drop  i32 . const 0  f32 . convert_i32_s  f64 . promote_f32  drop  i32 . const 0  i32 . extend8_s  drop  i32 . const 0  i32 . extend16_s  drop  i64 . const 0  i64 . extend8_s  drop  i64 . const 0  i64 . extend16_s  drop  i64 . const 0  i64 . extend32_s  drop  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: .EmptyStmts , mems: .EmptyStmts , globals: .EmptyStmts , elem: .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
